### PR TITLE
Added Cumprod Instance method to Jax NumPy Frontend

### PR DESCRIPTION
**Pull request by RakshitKumar04**

### CODE REVIEW

**Comment by RakshitKumar04** [comment]
Close #14954

**Comment by RakshitKumar04** [comment]
Hey, @AnnaTz Please review my PR.

**Comment by RakshitKumar04** [comment]
Hey, @AnnaTz I am facing this `Assertion Error` in my implementation. Can you please guide me on how to resolve this.

```
E       AssertionError: the ground truth framework jax returned a int64 datatype while the backend paddle returned a int32 datatype
E       Falsifying example: test_jax_devicearray_cumprod(
E           dtype_and_x=(['int64'], [array([0])], 0),
E           method_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E           ),
E           init_flags=FrontendMethodTestFlags(
E               num_positional_args=0,
E               as_variable=[False],
E               native_arrays=[False],
E           ),
E           frontend_method_data=FrontendMethodData(ivy_init_module=<module 'ivy.functional.frontends.jax.numpy' from '/ivy/ivy/functional/frontends/jax/numpy/__init__.py'>, framework_init_module=<module 'jax.numpy' from '/usr/local/lib/python3.8/dist-packages/jax/numpy/__init__.py'>, init_name='array', method_name='cumprod'),
E           on_device='cpu',
E           frontend='jax',
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.70.2', b'AXicY2BkQAUAACEAAg==') as a decorator on your test case
```

File: ivy/functional/frontends/jax/devicearray.py [hunk 1]
```diff
@@ -73,6 +73,16 @@ def mean(self, *, axis=None, dtype=None, out=None, keepdims=False, where=None):
             keepdims=keepdims,
             where=where,
         )
+    
+    def cumprod(self, axis=None, dtype=None, out=None):
+        if dtype is None:
+            dtype = ivy.as_ivy_dtype(self.dtype)
+        return jax_frontend.numpy.cumsum(
```
Comment (AnnaTz) [inline ivy/functional/frontends/jax/devicearray.py:80]: Did you use `cumsum` instead of `cumprod` by mistake here?

Comment (AnnaTz) [reply ivy/functional/frontends/jax/devicearray.py:80]: As for the dtype error, I think you should first check to see if the frontend function you are using passes its test.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/jax/devicearray.py (hunk 1 above)
Comment (RakshitKumar04) [reply ivy/functional/frontends/jax/devicearray.py:80]: Ohhh sorry, as I have sent PR for `cumsum` and `cumprod` both, that somehow got mixed. And I think that was the reason I was receiving the error.

**Comment by RakshitKumar04** [comment]
Hey, @AnnaTz I have corrected the implementation and it is passing 4 out of 5 tests. 
Please take a look.

**Comment by AnnaTz** [comment]
> Hey, @AnnaTz I have corrected the implementation and it is passing 4 out of 5 tests. Please take a look.

I don't see any errors in the CI check. Is it failing in the paddle backend only? And if so, what's the error?

**Comment by RakshitKumar04** [comment]
@AnnaTz I don't know why, but when I locally ran the tests yesterday, one of the tests didn't pass but today all 5 got passed, Sorry

**Comment by AnnaTz** [comment]
> @AnnaTz I don't know why, but when I locally ran the tests yesterday, one of the tests didn't pass but today all 5 got passed, Sorry

No worries, looks good to me! I'll merge this.

**Review by AnnaTz — APPROVED** [review]

**Pull request merged**